### DISCUSSION TITLE
Strongswan says IKE SA established, the peer says received Plain text authentication failed

### DISCUSSION THREAD

**Discussion by cyanideangel2000** [strongswan/strongswan#1167] [Q&A]
Hello Team,

I am running into this interesting scenario where the two peers end up trying for eternity until intervened and have different opinion about Auth status
The peer that i can probe into is the linux box with strongswan 5.7.2.
The other peer is a network equipment vendor.

Tunnel comes up and things hold okay.
Then, the vendor detects a DPD failure and immediately removes the SAs at its end.
It then re-initiates the IKE and continues to try to get the tunnel up until success

My end is the strongswan.
Since I don't have DPD enabled, my end doesn't notice the DPD failure.
However it starts to receive the IKE INIT messages and also have unique set as "replace" to keep a single IKE session active to a single id peer.

Usual exchange follows and strongswan logs

```
[NET] <99335> received packet: from PEER-IP[500] to My-IP[500] (469 bytes)
[IKE]   <99335> PEER-IP is initiating an IKE_SA
[NET] <99335> sending packet: from My-IP[500] to PEER-IP[500] (448 bytes)
[NET] <99335> received packet: from PEER-IP[4500] to My-IP[4500] (288 bytes)
[CFG] <99335> looking for peer configs matching My-IP[My-IP]...PEER-IP[PEER-IP]
[IKE] <VendorTunnel1|99335> authentication of 'PEER-IP' with pre-shared key successful
[IKE] <VendorTunnel1|99294> schedule delete of duplicate IKE_SA for peer 'PEER-IP' due to uniqueness policy and suspected reauthentication
[IKE] <VendorTunnel1|99335> IKE_SA VendorTunnel1[99335] established between My-IP[My-IP]...PEER-IP[PEER-IP]
[IKE] <VendorTunnel1|99294> deleting IKE_SA tenant-12523-4[99294] between My-IP[My-IP]...PEER-IP[PEER-IP]
```

However vendor reports
```
IKE-Event: IKEv2 packet R(PEER-IP:4500 <- My-IP:4500): len=   60, mID=1, HDR, N(AUTHENTICATION_FAILED)
IKE-Event: Plain-text notification `Authentication failed' (24) from My-IP:4500 for protocol None. Initiator SPI 02003b43 d2f7fdae Responder SPI 06c2b528 b990a930
```

This continues until admin intervene and clear out the IKE SAs at the Vendor end.
My Question:
1. Why is that at strongswan IKE SA gets ESTAB when strongswan is sending an AUTH FAIL notify
2. Is there a config at strongswan that can help avoid running into this issue. For us we have run into this handful of times. Will closeikeonerror good enough to take care of this?
3. I don't see the local authentication log at strongswan end still the IKE SA gets established? Only the PEER-IP/PEER-ID gets authenticated
4. The config unique = replace has any role to play here?

Thank you in advance.

**Comment by tobiasbrunner** [comment]
> Why is that at strongswan IKE SA gets ESTAB when strongswan is sending an AUTH FAIL notify

Does it? You haven't provided any log that shows that (please see #196).

**Reply by cyanideangel2000** [reply]
The following log from the description helps?
`[IKE] <VendorTunnel1|99335> IKE_SA VendorTunnel1[99335] established between My-IP[My-IP]...PEER-IP[PEER-IP]`

**Reply by tobiasbrunner** [reply]
It doesn't.

**Comment by owensteam** [comment]
could you get more verbose logs?
